cm.py

```diff
@@ -125,11 +125,25 @@ def parcel(self, product=None, version=None, stage=None):
         if not product and not version and not stage:
             raise Exception('A product and/or version and/or stage must '
                             'be specified to select a parcel.')
-        return next(parcel
+        return next((parcel
                     for parcel in self.parcels
                     if (not product or parcel.product == product)
                     and (not version or parcel.version == version)
-                    and (not stage or parcel.stage == stage))
+                    and (not stage or parcel.stage == stage)), None)
+
+    def wait_for_parcel_availability(self, product, version=None, stage=None):
+        def condition(product, version, stage):
+            parcel = self.parcel(product=product, version=version, stage=stage)
+            return parcel is not None
+
+        def success(time):
+            logger.debug('%s parcel with %s version found in %s stage after %s seconds.', product, version, stage, time)
+
+        def failure(timeout):
+            raise TimeoutError('Timed out after {} seconds waiting for %s parcel with %s version'
```

Comment:
I think you're mixing {} with %s in the `format` string? :)

Reply:
Done.